.github/workflows/ant.yml

```diff
@@ -9,10 +9,10 @@ jobs:
 
     steps:
     - uses: actions/checkout@v3
-    - name: Set up JDK 1.8
+    - name: Set up JDK 9
       uses: actions/setup-java@v3
       with:
-        java-version: '8'
+        java-version: '9'
```

Comment:
@olby sorry for late response. I don't think removing the 'b' will help - see https://docs.oracle.com/javase%2F9%2Fdocs%2Fapi%2F%2F/java/lang/module/ModuleFinder.html#of-java.nio.file.Path...-

> If the name matches the regular expression "-(\\d+(\\.|$))" then the module name will be derived from the subsequence preceding the hyphen of the first occurrence. The subsequence after the hyphen is parsed as a [Version](https://docs.oracle.com/javase%2F9%2Fdocs%2Fapi%2F%2F/java/lang/module/ModuleDescriptor.Version.html) and ignored if it cannot be parsed as a Version.

The error is however:

```
cause: java.cup.runtime.11b.20160615: Invalid module name: '11b' is not a Java identifier
```

It's because automatic module deriving did not parse `11b-20160615-2` as a version and so `11b` (as well as `20160615`) is considered as part of the module name. The "version" part is only the last `2` after the last hyphen.

So I think instead of manipulating with the version, or moving to Java 9+, we can add `Automatic-Module-Name` to the manifest file instead:

> If the JAR file has the attribute "Automatic-Module-Name" in its main manifest then its value is the [module name](https://docs.oracle.com/javase%2F9%2Fdocs%2Fapi%2F%2F/java/lang/module/ModuleDescriptor.html#name--). The module name is otherwise derived from the name of the JAR file.

Reply:
I did necessary changes in version `11b-20160615-3`. Can you please try?